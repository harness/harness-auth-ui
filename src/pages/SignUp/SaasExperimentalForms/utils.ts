/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import CDLogo from "static/images/cd-logo.svg";
import CILogo from "static/images/ci-logo.svg";
import CCMLogo from "static/images/ccm-logo.svg";
import FFLogo from "static/images/ff-logo.svg";
import ChaosLogo from "static/images/chaos-logo.svg";
import HarnessLogo from "static/images/logo.svg";

import PathCD from "static/images/path-cd.svg";
import PathCI from "static/images/path-ci.svg";
import PathCCM from "static/images/path-ccm.svg";
import PathFF from "static/images/path-ff.svg";
import PathChaos from "static/images/path-chaos.svg";
import PathHarness from "static/images/path-harness.svg";

interface ModuleDetailsMap {
  [key: string]: ModuleDetail;
}

interface ModuleDetail {
  logo: string;
  pathImg: string;
  title: string;
  tagLine: string;
  valueProp: string;
  callout: string;
  intro: string;
  cta: string;
  color: string;
}
const MODULE_TO_DETAILS_MAP: ModuleDetailsMap = {
  cd: {
    logo: CDLogo,
    pathImg: PathCD,
    title: "Continuos Delivery",
    tagLine: "Blazing Fast Deployment Pipelines in Minutes",
    valueProp:
      "Self-Service Continuous Delivery that enables engineers to deploy on-demand, without a single script",
    callout:
      "Truatsed by 1000s of Developers, driving 17 Million deployments in last 12 months",
    intro:
      "Define Pipelines that enable you to improve your deployment times, manage all in one place",
    cta: "Let's Deploy",
    color: "#5FB34E"
  },
  chaos: {
    logo: ChaosLogo,
    pathImg: PathChaos,
    title: "Chaos Engineering",
    tagLine: "Blazing Fast Deployment Pipelines in Minutes",
    valueProp:
      "Self-Service Continuous Delivery that enables engineers to deploy on-demand, without a single script",
    callout:
      "Trusted by 1000s of Developers, driving 17 Million deployments in last 12 months",
    intro:
      "Define Pipelines that enable you to improve your deployment times, manage all in one place",
    cta: "Let's Deploy",
    color: "#FF006A"
  },
  ci: {
    logo: CILogo,
    pathImg: PathCI,
    title: "Continuos Integration",
    tagLine: "Harness CI is an enterprise ready, cloud native CI product.",
    valueProp:
      "An intelligent, container-native CI solution with isolated builds and standardized extensions. Build artifacts smarter and faster.",
    callout:
      "Trusted by the New York Times, Roblox, John Deere, eBay and many others to run their build and test suits balzing fast",
    intro:
      "Create a container-based build pipeline in minutes. Even testing is lightweight and fast with Test Intelligence!",
    cta: "Let's build",
    color: "#2BB1F2"
  },
  ccm: {
    logo: CCMLogo,
    pathImg: PathCCM,
    title: "Cloud Cost Management",
    tagLine: "Gain complete cloud control and automated cost optimization ",
    valueProp:
      "Achieve cost transparency and reclaim wasted spend across clouds - with built-in anomaly detection and automation",
    callout:
      "Trusted by developers and finance teams from leading startups to enterprise",
    intro:
      "Manage cloud costs and automate savings from kubernetes pods to databases",
    cta: "Connect Cloud Account",
    color: "#01C9CC"
  },
  ff: {
    logo: FFLogo,
    pathImg: PathFF,
    title: "Feature Flags",
    tagLine: "Confidently deploy features with both velocity and control.",
    valueProp:
      "Ship more features and reduce deployment risk with the control and governance enterprises require - all without breaking things, even while in production.",
    callout: "Trusted by Developers and Product teams from zero to scale.",
    intro:
      "Automate feature releases, build with flags as code, and ship more features with less risk.",
    cta: "Start Managing Features",
    color: "#EE8625"
  },
  default: {
    logo: HarnessLogo,
    pathImg: PathHarness,
    title: "Harness",
    tagLine: "Loved by Developers, Trusted by Businesses",
    valueProp:
      "Harness is the industry's first Software Delivery Platform to use AI to simplify your DevOps processes - CI, CD, Feature Flags, Cloud Costs, and much more.",
    callout:
      "Trusted by Developers, DevOps, DevSecOps, Finance and others, across world's leading digital businesses.",
    intro:
      "Define Pipelines that enable you to improve your deployment times, manage all in one place",
    cta: "Let's Deploy",
    color: "#0278D5"
  }
};

const getModuleDetails = (moduleName: string): ModuleDetail => {
  const details = MODULE_TO_DETAILS_MAP[moduleName];
  if (details) {
    return details;
  }
  return MODULE_TO_DETAILS_MAP.default;
};

export { getModuleDetails };
