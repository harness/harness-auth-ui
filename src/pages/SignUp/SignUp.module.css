.signup {
  & .header {
    display: flex;
    margin-bottom: var(--spacing-huge);
    align-items: center;
    font-size: var(--font-size-medium);
  }

  & .title {
    font-size: 32px;
    font-weight: 600;
    margin-bottom: var(--spacing-small);
  }

  & .subtitle {
    font-size: var(--font-size-medium);
    margin-bottom: var(--spacing-huge);
    color: var(--dark-400);
  }

  & .footer {
    margin-top: var(--spacing-huge);
  }
}
