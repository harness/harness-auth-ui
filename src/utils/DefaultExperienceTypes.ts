export enum DefaultExperience {
  NG = "NG",
  CG = "CG"
}
