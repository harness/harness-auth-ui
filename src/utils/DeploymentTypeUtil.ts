export const isCommunityPlan = (): boolean =>
  window.deploymentType === "COMMUNITY";

export const isOnPrem = (): boolean => window.deploymentType === "ON_PREM";

export const isSaas = (): boolean => window.deploymentType === "SAAS";
