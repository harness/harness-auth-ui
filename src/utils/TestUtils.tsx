import React from "react";
import qs from "qs";
import { compile } from "path-to-regexp";
import { Router, Route, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";
import { RestfulProvider, UseGetProps } from "restful-react";
import { queryByAttribute } from "@testing-library/react";

export interface TestWrapperProps {
  path?: string;
  pathParams?: Record<string, string | number>;
  queryParams?: Record<string, unknown>;
}

export const TestWrapper: React.FC<TestWrapperProps> = (props) => {
  const { path = "/", pathParams = {}, queryParams = {} } = props;

  const search = qs.stringify(queryParams, { addQueryPrefix: true });
  const routePath = compile(path)(pathParams) + search;

  const history = React.useMemo(
    () => createMemoryHistory({ initialEntries: [routePath] }),
    []
  );

  return (
    <Router history={history}>
      <RestfulProvider base="/">
        <Switch>
          <Route exact path={path}>
            {props.children}
          </Route>
        </Switch>
      </RestfulProvider>
    </Router>
  );
};

export const queryByNameAttribute = (
  name: string,
  container: HTMLElement
): HTMLElement | null => queryByAttribute("name", container, name);

export type UseGetMockData<
  TData,
  TError = undefined,
  TQueryParams = undefined,
  TPathParams = undefined
> = Required<UseGetProps<TData, TError, TQueryParams, TPathParams>>["mock"];
