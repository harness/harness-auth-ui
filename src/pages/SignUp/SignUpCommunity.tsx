import React from "react";
import cx from "classnames";
import { Link } from "react-router-dom";
import { Form } from "react-final-form";

import BasicLayout from "components/BasicLayout/BasicLayout";

import logo from "static/images/harness-logo.svg";
import css from "./SignUp.module.css";
import RouteDefinitions from "RouteDefinitions";
import Field from "components/Field/Field";
import { handleError } from "utils/ErrorUtils";
import { validateEmail, validatePassword } from "utils/FormValidationUtils";
import { useQueryParams } from "hooks/useQueryParams";
import { SignupDTO, useCommunitySignup } from "../../services/ng";
import { handleSignUpSuccess } from "../../utils/SignUpUtils";

interface SignUpFormData {
  email: string;
  password: string;
}

const SignUpCommunity: React.FC = () => {
  const { mutate: signup, loading } = useCommunitySignup({});
  const { module } = useQueryParams<{
    module?: string;
    signupAction?: string;
    edition?: string;
    billingFrequency?: string;
    utm_source?: string;
    utm_content?: string;
    utm_medium?: string;
    utm_term?: string;
    utm_campaign?: string;
  }>();

  const handleSignup = async (data: SignUpFormData): Promise<void> => {
    try {
      const signupRequestData: SignupDTO = {
        email: data.email,
        password: data.password,
        intent: module,
        edition: "COMMUNITY",
        signupAction: "REGULAR"
      };

      const handleCommunitySignup = await signup(signupRequestData);
      if (handleCommunitySignup.resource) {
        handleCommunitySignup.resource.intent = "COMMUNITY";
      }
      handleSignUpSuccess(handleCommunitySignup?.resource);
    } catch (error) {
      handleError(error);
    }
  };

  const onSubmit = (data: SignUpFormData) => {
    data.email = data.email.toLowerCase();
    handleSignup(data);
  };

  const emailField = (
    <Field
      name="email"
      label={"Email"}
      placeholder="email@work.com"
      disabled={loading}
      validate={validateEmail}
    />
  );

  const passwordField = (
    <Field
      name="password"
      label="Password"
      type="password"
      placeholder="Password"
      disabled={loading}
      validate={validatePassword}
    />
  );

  return (
    <BasicLayout>
      <div className={cx(css.signup)}>
        <div className={css.header}>
          <img src={logo} width={120} className={css.logo} />
        </div>
        <div className={css.title}>Sign up</div>
        <div className={css.subtitle}>and get ship done.</div>
        <Form
          onSubmit={onSubmit}
          render={({ handleSubmit }) => (
            <form
              className="layout-vertical spacing-medium"
              onSubmit={handleSubmit}
            >
              {emailField}
              {passwordField}
              <input
                type="submit"
                value="Sign up"
                className="button primary"
                disabled={loading}
              />
            </form>
          )}
        />
        <div className={css.footer}>
          Already have an account?{" "}
          <Link to={RouteDefinitions.toSignIn()}>Sign in</Link>
        </div>
      </div>
    </BasicLayout>
  );
};

export default SignUpCommunity;
