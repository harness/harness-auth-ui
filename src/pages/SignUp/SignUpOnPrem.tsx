/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from "react";
import cx from "classnames";
import { Form } from "react-final-form";
import { useSignup } from "services/portal";
import { Link, useHistory } from "react-router-dom";

import BasicLayout from "components/BasicLayout/BasicLayout";

import Field from "components/Field/Field";
import { handleError } from "utils/ErrorUtils";
import { validateEmail, validateString } from "utils/FormValidationUtils";
import PasswordField from "components/Field/PasswordField";
import { URLS } from "interfaces/OAuthProviders";
import Text from "components/Text/Text";
import RouteDefinitions from "RouteDefinitions";

import logo from "static/images/harness-logo.svg";
import css from "./SignUp.module.css";
import toast from "react-hot-toast";

interface SignUpFormData {
  email: string;
  password: string;
  accountName: string;
  companyName: string;
  firstname?: string;
  lastname?: string;
}

const SignUpOnPrem: React.FC = () => {
  const { mutate: signup, loading } = useSignup({ source: "ONPREM" });
  const history = useHistory();

  const handleSignup = async (formData: SignUpFormData): Promise<void> => {
    try {
      const payload = {
        ...formData,
        name: `${formData.firstname} ${formData.lastname}`
      };
      delete payload.firstname;
      delete payload.lastname;
      // `any` is required because BE is incorrectly using Invite class for signup
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const signupResponse = await signup(payload as any);
      if (signupResponse) {
        toast.success("Account created successfully!");
        history.push(RouteDefinitions.toSignIn());
      } else {
        toast.error("Something went wrong");
      }
    } catch (error) {
      handleError(error);
    }
  };

  const onSubmit = (data: SignUpFormData) => {
    data.email = data.email.toLowerCase();
    handleSignup(data);
  };

  return (
    <BasicLayout>
      <div className={cx(css.signup)}>
        <div className={css.header}>
          <img src={logo} width={120} className={css.logo} />
          <div style={{ flex: 1 }} />
          <Link to={RouteDefinitions.toSignIn()}>
            <Text icon="leftArrow">Sign in</Text>
          </Link>
        </div>
        <div className={css.title}>Welcome to Harness</div>
        <div className={css.communitySubtitle}>Set up an admin user</div>
        <Form
          onSubmit={onSubmit}
          render={({ handleSubmit }) => (
            <form
              className="layout-vertical spacing-medium"
              onSubmit={handleSubmit}
            >
              <Field
                name="email"
                label={"Email"}
                placeholder="email@work.com"
                disabled={loading}
                validate={validateEmail}
              />
              <PasswordField
                name="password"
                label="Password"
                placeholder="Password"
                disabled={loading}
              />
              <Field
                name="accountName"
                label={"Account Name (Optional)"}
                placeholder="Account"
                disabled={loading}
              />
              <Field
                name="companyName"
                label={"Company Name"}
                placeholder="Company"
                disabled={loading}
                validate={validateString}
              />
              <Field
                name="firstname"
                label={"First Name"}
                placeholder="John"
                disabled={loading}
                validate={validateString}
              />
              <Field
                name="lastname"
                label={"Last Name"}
                placeholder="Smith"
                disabled={loading}
                validate={validateString}
              />
              <input
                type="submit"
                value={loading ? "Creating user..." : "Create user"}
                className="button primary"
                disabled={loading}
              />
            </form>
          )}
        />
        <div className={cx(css.disclaimer)}>
          By registering, you agree to our&nbsp;
          <a
            className={css.externalLink}
            href={URLS.PRIVACY_AGREEMENT}
            rel="noreferrer"
            target="_blank"
          >
            Privacy Policy&nbsp;
          </a>
          and our&nbsp;
          <a
            className={css.externalLink}
            href={URLS.SUBSCRIPTION_TERMS}
            rel="noreferrer"
            target="_blank"
          >
            Terms of Use.&nbsp;
          </a>
        </div>
      </div>
    </BasicLayout>
  );
};

export default SignUpOnPrem;
