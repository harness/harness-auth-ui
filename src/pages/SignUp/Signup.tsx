/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useRef, useState, FocusEvent } from "react";
import cx from "classnames";
import { Link, useHistory } from "react-router-dom";
import { Form } from "react-final-form";
import ReCAPTCHA from "react-google-recaptcha";

import BasicLayout from "components/BasicLayout/BasicLayout";
import { useSignup, SignupDTO } from "services/ng";

import logo from "static/images/harness-logo.svg";
import css from "./SignUp.module.css";
import RouteDefinitions from "RouteDefinitions";
import AuthFooter, { AuthPage } from "components/AuthFooter/AuthFooter";
import Field from "components/Field/Field";
import { handleError } from "utils/ErrorUtils";
import { validateEmail, validatePassword } from "utils/FormValidationUtils";
import telemetry from "telemetry/Telemetry";
import { useQueryParams } from "hooks/useQueryParams";
import { VERIFY_EMAIL_STATUS } from "pages/VerifyEmail/VerifyEmailStatus";
import { BillingFrequency, Edition, SignupAction } from "utils/SignUpUtils";

interface SignUpFormData {
  email: string;
  password: string;
}

const SignUp: React.FC = () => {
  const [captchaToken, setCaptchaToken] = useState<string | null>(null);
  const history = useHistory();
  const [signupData, setSignupData] = useState({ email: "", password: "" });
  const { mutate: signup, loading } = useSignup({});
  const captchaRef = useRef<ReCAPTCHA>(null);
  const {
    module,
    signupAction,
    edition,
    billingFrequency,
    utm_source,
    utm_content,
    utm_medium,
    utm_term,
    utm_campaign
  } = useQueryParams<{
    module?: string;
    signupAction?: string;
    edition?: string;
    billingFrequency?: string;
    utm_source?: string;
    utm_content?: string;
    utm_medium?: string;
    utm_term?: string;
    utm_campaign?: string;
  }>();

  const [captchaExecuting, setCaptchaExecuting] = useState(false);
  useEffect(() => {
    const { email, password } = signupData;

    if (email && password && captchaToken) {
      setCaptchaExecuting(false);
      handleSignup(signupData, captchaToken);
    }
  }, [captchaToken]);

  const handleSignup = async (
    data: SignUpFormData,
    captchaToken: string
  ): Promise<void> => {
    const encodedEmail = encodeURIComponent(data.email);
    try {
      const signupRequestData: SignupDTO = {
        ...data,
        intent: module,
        utmInfo: {
          utmSource: utm_source,
          utmContent: utm_content,
          utmMedium: utm_medium,
          utmTerm: utm_term,
          utmCampaign: utm_campaign
        }
      };

      if (signupAction && signupAction.toUpperCase() in SignupAction) {
        signupRequestData.signupAction = signupAction.toUpperCase() as SignupAction;
      }

      if (edition && edition.toUpperCase() in Edition) {
        signupRequestData.edition = edition.toUpperCase() as Edition;
      }

      if (
        billingFrequency &&
        billingFrequency.toUpperCase() in BillingFrequency
      ) {
        signupRequestData.billingFrequency = billingFrequency.toUpperCase() as BillingFrequency;
      }

      await signup(signupRequestData, {
        queryParams: { captchaToken: captchaToken }
      });

      history.push({
        pathname: RouteDefinitions.toEmailVerification(),
        search: `?status=${VERIFY_EMAIL_STATUS.EMAIL_SENT}&email=${encodedEmail}`
      });
    } catch (error) {
      captchaRef.current?.reset();

      if (
        error?.data?.responseMessages?.length &&
        error?.data?.responseMessages[0]?.code === "USER_ALREADY_REGISTERED"
      ) {
        history.push({
          pathname: RouteDefinitions.toEmailVerification(),
          search: `?status=${VERIFY_EMAIL_STATUS.SIGNED_UP}&email=${encodedEmail}`
        });
      } else {
        handleError(error);
      }
    }
  };

  const manuallyExcecuteRecaptcha = (): boolean => {
    if (captchaRef.current?.execute) {
      captchaRef.current.execute();
      setCaptchaExecuting(true);
      return true;
    }

    handleError("Captcha failed to execute");
    return false;
  };

  const onSubmit = (data: SignUpFormData) => {
    if (manuallyExcecuteRecaptcha()) {
      data.email = data.email.toLowerCase();
      setSignupData(data);
      telemetry.track({
        event: "Signup submit",
        properties: {
          category: "SIGNUP",
          userId: data.email,
          groupId: "",
          utm_source: utm_source || "",
          utm_medium: utm_medium || "",
          utm_campaign: utm_campaign || "",
          utm_term: utm_term || "",
          utm_content: utm_content || ""
        }
      });
    }
  };

  const emailField = (
    <Field
      name="email"
      label={"Email"}
      placeholder="email@work.com"
      disabled={loading || captchaExecuting}
      validate={validateEmail}
      onBlur={(e: FocusEvent<HTMLInputElement>) => {
        telemetry.track({
          event: "Email input",
          properties: {
            category: "SIGNUP",
            userId: e.target.value,
            groupId: "",
            utm_source: utm_source || "",
            utm_medium: utm_medium || "",
            utm_campaign: utm_campaign || "",
            utm_term: utm_term || "",
            utm_content: utm_content || ""
          }
        });
      }}
    />
  );

  const passwordField = (
    <Field
      name="password"
      label="Password"
      type="password"
      placeholder="Password"
      disabled={loading || captchaExecuting}
      validate={validatePassword}
      infoMessage={"The password must be between 8 and 64 characters long"}
    />
  );

  telemetry.page({
    name: "Signup Page",
    category: "SIGNUP",
    properties: {
      userId: "",
      groupId: "",
      module: module || "",
      utm_source: utm_source || "",
      utm_medium: utm_medium || "",
      utm_campaign: utm_campaign || "",
      utm_term: utm_term || "",
      utm_content: utm_content || ""
    }
  });

  function handleRecaptchaError() {
    // Block the user until they refresh
    setCaptchaExecuting(true);
    handleError("Captcha has failed, please refresh the page.");
  }

  return (
    <BasicLayout>
      <div className={cx(css.signup)}>
        <div className={css.header}>
          <img src={logo} width={120} className={css.logo} />
        </div>
        <div className={css.title}>Sign up</div>
        <div className={css.subtitle}>and get ship done.</div>
        <Form
          onSubmit={onSubmit}
          render={({ handleSubmit }) => (
            <form
              className="layout-vertical spacing-medium"
              onSubmit={handleSubmit}
            >
              {emailField}
              {passwordField}
              <ReCAPTCHA
                ref={captchaRef}
                sitekey={
                  window.invisibleCaptchaToken ||
                  "6LfMgvcaAAAAAHCctQKV5AsCYZJHsKOpGH5oGc5Q" // site key for dev environments
                }
                size="invisible"
                onChange={setCaptchaToken}
                onErrored={handleRecaptchaError}
              />
              <input
                type="submit"
                value="Sign up"
                className="button primary"
                disabled={loading || captchaExecuting}
              />
            </form>
          )}
        />
        <AuthFooter page={AuthPage.SignUp} />
        <div className={css.footer}>
          Already have an account?{" "}
          <Link to={RouteDefinitions.toSignIn()}>Sign in</Link>
        </div>
      </div>
    </BasicLayout>
  );
};

export default SignUp;
