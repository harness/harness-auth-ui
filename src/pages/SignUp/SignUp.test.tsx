import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";

import SignUp from "./Signup";
import { TestWrapper, queryByNameAttribute } from "utils/TestUtils";

const pageMock = jest.fn();
const trackMock = jest.fn();

jest.mock("telemetry/Telemetry", () => ({
  page: jest.fn().mockImplementation((values) => pageMock(values)),
  track: jest.fn().mockImplementation((values) => trackMock(values))
}));

describe("SignUp", () => {
  test("render", () => {
    const { container } = render(
      <TestWrapper path={"/signup"}>
        <SignUp />
      </TestWrapper>
    );
    expect(container).toMatchSnapshot();
  });

  test("telemetry", async () => {
    const { container, getByRole } = render(
      <TestWrapper path={"/signup"} queryParams={{ module: "ci" }}>
        <SignUp />
      </TestWrapper>
    );
    await waitFor(() =>
      expect(pageMock).toBeCalledWith({
        name: "Signup Page",
        category: "SIGNUP",
        properties: {
          userId: "",
          groupId: "",
          module: "ci"
        }
      })
    );
    fireEvent.input(queryByNameAttribute("email", container)!, {
      target: { value: "random@hotmail.com" },
      bubbles: true
    });
    fireEvent.focusOut(queryByNameAttribute("email", container)!);
    await waitFor(() =>
      expect(trackMock).toBeCalledWith({
        event: "Email input",
        properties: {
          category: "SIGNUP",
          userId: "random@hotmail.com",
          groupId: ""
        }
      })
    );
    fireEvent.input(queryByNameAttribute("password", container)!, {
      target: { value: "12345678" },
      bubbles: true
    });
    fireEvent.click(getByRole("button", { name: "Sign up" }));

    await waitFor(() =>
      expect(trackMock).toBeCalledWith({
        event: "Signup submit",
        properties: {
          category: "SIGNUP",
          userId: "random@hotmail.com",
          groupId: ""
        }
      })
    );
    expect(container).toMatchSnapshot();
  });
});
