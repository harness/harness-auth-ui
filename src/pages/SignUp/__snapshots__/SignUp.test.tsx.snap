// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp render 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="cardColumn"
    >
      <div
        class="card"
      >
        <div
          class="signup"
        >
          <div
            class="header"
          >
            <img
              class="logo"
              src="test-file-stub"
              width="120"
            />
          </div>
          <div
            class="title"
          >
            Sign Up
          </div>
          <div
            class="subtitle"
          >
            and get ship done.
          </div>
          <form
            class="layout-vertical spacing-medium"
          >
            <div
              class="layout-vertical spacing-small field"
            >
              <label>
                Email
              </label>
              <input
                class=""
                id="email"
                name="email"
                placeholder="email@work.com"
                value=""
              />
            </div>
            <div
              class="layout-vertical spacing-small field"
            >
              <label>
                Password
              </label>
              <input
                class=""
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div />
            <input
              class="button primary"
              type="submit"
              value="Sign Up"
            />
          </form>
          <h2
            class="lineMessage"
          >
            <span
              class="message"
            >
              or sign up with
            </span>
          </h2>
          <div
            class="layout-horizontal spacing-auto oAuthIcons"
          >
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=github&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=bitbucket&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=gitlab&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=linkedin&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=google&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=azure&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
          </div>
          <div
            class="disclaimer"
          >
            By signing up, you agree to our 
            <a
              class="externalLink"
              href="https://harness.io/privacy/"
              rel="noreferrer"
              target="_blank"
            >
              Privacy Policy 
            </a>
            and our 
            <a
              class="externalLink"
              href="https://harness.io/subscriptionterms/"
              rel="noreferrer"
              target="_blank"
            >
              Terms of Use
            </a>
          </div>
          <div
            class="footer"
          >
            Already have an account?
             
            <a
              href="/signin"
            >
              Sign In
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="imageColumn"
    >
      <img
        alt=""
        aria-hidden="true"
        class="image"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;

exports[`SignUp telemetry 1`] = `
<div>
  <div
    class="layout"
  >
    <div
      class="cardColumn"
    >
      <div
        class="card"
      >
        <div
          class="signup"
        >
          <div
            class="header"
          >
            <img
              class="logo"
              src="test-file-stub"
              width="120"
            />
          </div>
          <div
            class="title"
          >
            Sign Up
          </div>
          <div
            class="subtitle"
          >
            and get ship done.
          </div>
          <form
            class="layout-vertical spacing-medium"
          >
            <div
              class="layout-vertical spacing-small field"
            >
              <label>
                Email
              </label>
              <input
                class=""
                disabled=""
                id="email"
                name="email"
                placeholder="email@work.com"
                value="random@hotmail.com"
              />
            </div>
            <div
              class="layout-vertical spacing-small field"
            >
              <label>
                Password
              </label>
              <input
                class=""
                disabled=""
                id="password"
                name="password"
                placeholder="Password"
                type="password"
                value="12345678"
              />
            </div>
            <div />
            <input
              class="button primary"
              disabled=""
              type="submit"
              value="Sign Up"
            />
          </form>
          <h2
            class="lineMessage"
          >
            <span
              class="message"
            >
              or sign up with
            </span>
          </h2>
          <div
            class="layout-horizontal spacing-auto oAuthIcons"
          >
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=github&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=bitbucket&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=gitlab&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=linkedin&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=google&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
            <a
              class="iconContainer"
              href="http://localhost/gateway/api/users/oauth2Redirect?provider=azure&action=signup&isNG=true"
            >
              <img
                height="24"
                src="test-file-stub"
                width="24"
              />
            </a>
          </div>
          <div
            class="disclaimer"
          >
            By signing up, you agree to our 
            <a
              class="externalLink"
              href="https://harness.io/privacy/"
              rel="noreferrer"
              target="_blank"
            >
              Privacy Policy 
            </a>
            and our 
            <a
              class="externalLink"
              href="https://harness.io/subscriptionterms/"
              rel="noreferrer"
              target="_blank"
            >
              Terms of Use
            </a>
          </div>
          <div
            class="footer"
          >
            Already have an account?
             
            <a
              href="/signin"
            >
              Sign In
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="imageColumn"
    >
      <img
        alt=""
        aria-hidden="true"
        class="image"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;
